opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1516,610 +2031,1764 @@ void CpGrid::addLgrsUpdateLeafView(const std::vector<std::array<int,3>>& cells_p
+                                                  const std::vector<int> refined_cell_count_vec,
+                                                  std::map<std::array<int,2>,int> elemLgrAndElemLgrCell_to_adaptedCell,
+                                                  std::unordered_map<int,std::array<int,2>> adaptedCell_to_elemLgrAndElemLgrCell,
+                                                  const int cell_count)
+{
+    const int& preAdaptMaxLevel = this->maxLevel();
+    const int& auxiliaryIdx = (preAdaptMaxLevel>0) ? (preAdaptMaxLevel +1) : 1;
+    // -- Adapted to {level, cell index in that level}  --
+    leaf_to_level_cells.resize(cell_count);
+    for (int cell = 0; cell < cell_count; ++cell) {
+        const auto& elemLgr = adaptedCell_to_elemLgrAndElemLgrCell[cell][0];
+        const auto& elemLgrCell = adaptedCell_to_elemLgrAndElemLgrCell[cell][1];
+        // elemLgr == -1 means that this adapted cell is equivalent to a cell from the starting grid. So we need to find out the level where that equivalent
+        // cell was born, as well as its cell index in that level.
+        if (elemLgr == -1) {
+            const auto& element = Dune::cpgrid::Entity<0>(*current_view_data_, elemLgrCell, true);
+            const auto& levelCellIdx = (element.level() == 0) ? element.getOrigin().index() : element.getLevelElem().index();
+            leaf_to_level_cells[cell] = { element.level(), levelCellIdx};
+        }
+        else {
+            const auto& levelWhereCellWasBorn = elemLgrAndElemLgrCell_to_refinedLevelAndRefinedCell[{elemLgr, elemLgrCell}][0];
+            const auto& levelCellIdx =  elemLgrAndElemLgrCell_to_refinedLevelAndRefinedCell[{elemLgr, elemLgrCell}][1];
+            leaf_to_level_cells[cell] = { levelWhereCellWasBorn, levelCellIdx};
         }
     }
-    for (int l = 1; l < num_patches +1; ++l) {
-        for (long unsigned int corner = 0; corner < level_to_leaf_corners[l].size(); ++corner) {
-            const auto& level_data = *(this->data_[l]);
-            leaf_corners[level_to_leaf_corners[l][corner]]
-                = level_data.geometry_.geomVector(std::integral_constant<int,3>()) -> get(corner);
+    // -- Refined to adapted cells --
+    for (int shiftedLevel = 0; shiftedLevel < static_cast<int>(refined_cell_count_vec.size()); ++shiftedLevel){
+        refined_level_to_leaf_cells_vec[shiftedLevel].resize(refined_cell_count_vec[shiftedLevel]);
+        for (int cell = 0; cell < refined_cell_count_vec[shiftedLevel]; ++cell) {
+            const auto& elemLgr = refinedLevelAndRefinedCell_to_elemLgrAndElemLgrCell[{shiftedLevel + auxiliaryIdx, cell}][0];
+            const auto& elemLgrCell = refinedLevelAndRefinedCell_to_elemLgrAndElemLgrCell[{shiftedLevel + auxiliaryIdx, cell}][1];
+            const auto& adaptedCellIdx = elemLgrAndElemLgrCell_to_adaptedCell[{elemLgr, elemLgrCell}];
+            refined_level_to_leaf_cells_vec[shiftedLevel][cell] = adaptedCellIdx;
```

Comment:
Would this simpler statement work?
```suggestion
            const auto& elemLgrCellPair = refinedLevelAndRefinedCell_to_elemLgrAndElemLgrCell[{shiftedLevel + auxiliaryIdx, cell}];
            const auto& adaptedCellIdx = elemLgrAndElemLgrCell_to_adaptedCell[elemLgrCellPair];
            refined_level_to_leaf_cells_vec[shiftedLevel][cell] = adaptedCellIdx;
```
```

refined_level_to_leaf_cells_vec[shiftedLevel][cell] = elemLgrAndElemLgrCell_to_adaptedCell[refinedLevelAndRefinedCell_to_elemLgrAndElemLgrCell[elemLgrCellPair]

Reply:
I thought the other version was slightly more readers-friendly. This line does simplify the code though